### ISSUE TITLE
Policy based VPN replaced with route. Cannot set mode via terraform

### ISSUE THREAD

**Issue by bogdandragoiu** [IBM-Cloud/terraform-provider-ibm#2277]
### Terraform Version
Terraform v0.14.7
+ provider registry.terraform.io/hashicorp/ibm v1.21.0

### Affected Resource(s)
ibm_is_vpn_gateway_connection

### Terraform Configuration Files
```hcl
resource "ibm_is_vpn_gateway_connection" "vpngw_3_cn" {
  name           = "vpngw-3"
  vpn_gateway    = ibm_is_vpn_gateway.vpngw_3.id
  local_cidrs    = [ibm_is_subnet.dmz_extern_subnet2.ipv4_cidr_block]
  peer_address   = "x.x.x.x"
  peer_cidrs     = ["y.y.y.y/24"]
  ike_policy     = ibm_is_ike_policy.ikev2_aes256_sha256_dh14_3600.id
  ipsec_policy   = ibm_is_ipsec_policy.ipsec_aes256_sha256_pfs_group_14_3600.id
  interval       = 15
  timeout        = 30
  action         = "restart"
  admin_state_up = true
  preshared_key  = "XXXXXXXXXXXXXXXXX"
}
```

### Expected Behavior
After upgrade to v1.21.0, the policy based VPN GW and connection should be kept

### Actual Behavior
I get prompted to recreate and not only does the public IP get lost, but the **mode** is changed to **route** and I cannot specify **policy**. What are we supposed to do?
```
  # module.vpc.ibm_is_vpn_gateway_connection.vpngw_3 must be replaced
-/+ resource "ibm_is_vpn_gateway_connection" "vpngw_3_cn" {
      ~ authentication_mode = "psk" -> (known after apply)
      ~ created_at          = "2020-11-27T12:01:42.085Z" -> (known after apply)
      ~ id                  = "[redacted]" -> (known after apply)
      ~ mode                = "policy" -> (known after apply)
        name                = "vpngw-3-cn"
      ~ related_crn         = "crn:v1:bluemix:public:is:eu-de:a/[redacted]::vpn:[redacted]" -> (known after apply)
      ~ resource_type       = "vpn_gateway_connection" -> (known after apply)
      ~ status              = "up" -> (known after apply)
      + tunnels             = (known after apply)
      ~ vpn_gateway         = "[redacted]" -> (known after apply) # forces replacement
        # (10 unchanged attributes hidden)
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Upgrade provider
2. `terraform plan`

### References
https://cloud.ibm.com/docs/ibm-cloud-provider-for-terraform?topic=ibm-cloud-provider-for-terraform-vpc-gen2-resources#vpn-gateway-connection-output

**Comment by hkantare** [comment]
As a workaround can you add lifecycle ignore changes on tunnels..
We already have fix ready and available in next release

**Comment by hkantare** [comment]
sorry for misunderstanding ..From the above snippet we see the VPNGatewayID is doing a froce replacement..

Have you destroyed and recreaetd the new VPN gateway?

As per VPC infrastructure some days back we use to have only policy based VPN gateway ..But now they introduced two types of gateways route and policy based gateways..By default as per API if user doesn't provide mode its a route based..

In VPN gateway resource we have argument "mode" to control this beahaviour..Can you update the tempalte of vpn_gateway to add mode="policy"

**Comment by bogdandragoiu** [comment]
Ah ok, I see now. I thought this was for the connection ,not the GW itself. I set it on the GW and now I do not get prompted for a replacement. Route based VPNGW also existed when I updated it, but policy was the default then, I see now that the default changing is what caused our issue.

Since I saw the change on the connection, I assumed the issue was there. It was not clear to me that the mode was inherited from the GW.

Setting the mode explicitly on the GW fixes my project. Thanks.
